### PR TITLE
FE: Detailed Card #18

### PR DESCRIPTION
**Pull request by uche-okoroafor**
FE: Detailed Card #18

### What this PR does (required):
-User interface for Card details #18 
-User can add, edit or remove items from the Card

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

https://user-images.githubusercontent.com/71725892/141935778-3ee71caf-7e7e-4a3b-9cf0-7d0ac8e2d51b.mp4

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Please resolve the merge conflict in this branch

File: client/src/mocks/mockCard.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+//     deadline: '2019-01-08T20:23',
+//     checklist: [{ item: 'Eat', isChecked: true, _id: "4" }],
+//     attachment: {
+//       imageName: 'image4',
+//       imageSource:
+//         'https://media-exp1.licdn.com/dms/image/C4D03AQEVYXv67l-1TA/profile-displayphoto-shrink_400_400/0/1599999564029?e=1642636800&v=beta&t=uV64EM58yJRZfnv7lAxQysobzhdLDwgl37UezXaGmxg',
+//       _id: "4",
+//     },
+//   },
+// ];
```
Comment (ezeilo-su) [inline client/src/mocks/mockCard.ts:155]: Please let's get rid of this unused code block.

File: client/src/pages/Board/Board.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  const closeCardDetails = () => {
+    setOpenDialog(false);
+  };
+  const [disableSetting, setDisableSetting] = useState(true);
+  const [displayAttachment, setDisplayAttachment] = useState(false);
+  const [displayChecklist, setDisplayChecklist] = useState(false);
+
+  useEffect(() => {
+    if (boards) {
+      // console.log(boards, 'boards');
```
Comment (ezeilo-su) [inline client/src/pages/Board/Board.tsx:45]: Let's remove this comment from here

File: client/src/pages/Board/Board.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      setTimeout(() => {
+        // SetDisplayedBoard(boards[0]);
```
Comment (ezeilo-su) [inline client/src/pages/Board/Board.tsx:47]: Remove this one too

File: client/src/pages/Board/Board.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      }, 1000);
+      console.log(boards[0], 'boards2');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [boards]);
+
+  const openCardDetails = (card: ICard, cardId: string, columnId: string, boardId: string): void => {
+    console.log(boards);
```
Comment (ezeilo-su) [inline client/src/pages/Board/Board.tsx:55]: Please remove console log :smile:

File: client/src/pages/Board/CardDialogComponents/CardAttachment.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { IconButton } from '@mui/material';
+import AttachFileOutlinedIcon from '@mui/icons-material/AttachFileOutlined';
+import useStyles from '../useStyles';
+import CloseIcon from '@mui/icons-material/Close';
+import { useState } from 'react';
+import { Box, Typography, TextField } from '@material-ui/core';
+import Button from '@mui/material/Button';
+import DialogContentText from '@mui/material/DialogContentText';
+import Stack from '@mui/material/Stack';
+// import { saveAttachment, deleteAttachment } from '../../../helpers/APICalls/cardApiCalls';
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardAttachment.tsx]: Let's remove this comment line

File: client/src/pages/Board/CardDialogComponents/CardAttachment.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  const [imageName, setImageName] = useState(attachment?.imageName);
+  const [imageSource, setImageSource] = useState(attachment?.imageSource);
+  const [displayInput, setDisplayInput] = useState(false);
+
+  const handleDeleteAttachment = async (): Promise<void> => {
+    // try {
+    //   const response = await deleteAttachment();
+    //   setImageSource('');
+    //   setImageName('');
+    // } catch (err) {}
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardAttachment.tsx]: Get rid of these comments

File: client/src/pages/Board/CardDialogComponents/CardAttachment.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  };
+  const handleSaveAttachment = async (): Promise<void> => {
+    // try {
+    //   const response = await saveAttachment();
+    // } catch (err) {}
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardAttachment.tsx]: Remove this one too

File: client/src/pages/Board/CardDialogComponents/CardAttachment.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    setDisplayInput(false);
+  };
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    console.log(e.target.files);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardAttachment.tsx]: We don't need this one. Please remove.

File: client/src/pages/Board/CardDialogComponents/CardChecklist.tsx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+import FactCheckOutlinedIcon from '@mui/icons-material/FactCheckOutlined';
+import useStyles from '../useStyles';
+import CloseIcon from '@mui/icons-material/Close';
+import { useState } from 'react';
+import { Box, Typography, TextField } from '@material-ui/core';
+import Button from '@mui/material/Button';
+import DialogContentText from '@mui/material/DialogContentText';
+import Checkbox from '@mui/material/Checkbox';
+import Stack from '@mui/material/Stack';
+// import { saveListItem, checkItem, deleteList } from '../../../helpers/APICalls/cardApiCalls';
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardChecklist.tsx]: Remove this comment line

File: client/src/pages/Board/CardDialogComponents/CardChecklist.tsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  const [newItem, setNewItem] = useState('');
+  const [displayInput, setDisplayInput] = useState(false);
+  const [submitting, setSubmitting] = useState(false);
+
+  const handleSaveCheckList = async (): Promise<void> => {
+    // try {
+    //   const response = await saveListItem();
+    //   setDisplayInput(false);
+    //   setNewItem('');
+    // } catch (err) {}
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardChecklist.tsx]: Let's get rid of this comment block

File: client/src/pages/Board/CardDialogComponents/CardChecklist.tsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+    setList([...list, { item: newItem, isChecked: false }]);
+  };
+
+  const handleCheck = async (id: string): Promise<void> => {
+    //   try {
+    //     const response = await checkItem();
+    //   } catch (err) {}
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardChecklist.tsx]: This one too

File: client/src/pages/Board/CardDialogComponents/CardChecklist.tsx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+  };
+
+  const handleDeleteList = async (id: string): Promise<void> => {
+    //   try {
+    //     const response = await deleteList();
+    //   } catch (err) {}
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardChecklist.tsx]: This one too

File: client/src/pages/Board/CardDialogComponents/CardComment.tsx [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+  useEffect(() => {
+    setNewComment(comment);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [boards]);
+
+  const handleSaveComment = async (): Promise<void> => {
+    setIsSubmitting(true);
+    saveCardItem('comment', newComment, ids).then((data) => {
+      if (data.error) {
+        // updateSnackBarMessage(data.error.message);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardComment.tsx]: Please remove comment

File: client/src/pages/Board/CardDialogComponents/CardComment.tsx [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+      } else if (data.success) {
+        updateBoards();
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+
+        // updateSnackBarMessage('An unexpected error occurred. Please try again');
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardComment.tsx]: Let's also remove this one

File: client/src/pages/Board/CardDialogComponents/CardComment.tsx [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+    });
+    setIsSubmitting(false);
+  };
+  const handleDeleteComment = async (): Promise<void> => {
+    setIsSubmitting(true);
+    deleteCardItem('comment', newComment, ids).then((data) => {
+      if (data.error) {
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        // updateCardContext(data.success);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardComment.tsx]: Also get rid of this one

File: client/src/pages/Board/CardDialogComponents/CardDeadline.tsx [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+  useEffect(() => {
+    setDate(deadline);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [boards]);
+
+  const handleSaveDeadline = async (): Promise<void> => {
+    setIsSubmitting(true);
+    saveCardItem('deadline', date, ids).then((data) => {
+      if (data.error) {
+        // updateSnackBarMessage(data.error.message);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDeadline.tsx]: Remove

File: client/src/pages/Board/CardDialogComponents/CardDeadline.tsx [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+      } else if (data.success) {
+        updateBoards();
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDeadline.tsx]: Please remove

File: client/src/pages/Board/CardDialogComponents/CardDeadline.tsx [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+        console.error({ data });
+
+        // updateSnackBarMessage('An unexpected error occurred. Please try again');
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDeadline.tsx]: Remove

File: client/src/pages/Board/CardDialogComponents/CardDeadline.tsx [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+  };
+  const handleDeleteDeadline = async (): Promise<void> => {
+    setIsSubmitting(true);
+
+    deleteCardItem('deadline', date, ids).then((data) => {
+      if (data.error) {
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDeadline.tsx]: Please remove

File: client/src/pages/Board/CardDialogComponents/CardDescription.tsx [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+  useEffect(() => {
+    setNewDescription(description);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [boards]);
+
+  const handleSaveDescription = async (): Promise<void> => {
+    saveCardItem('description', newDescription, ids).then((data) => {
+      setIsSubmitting(true);
+      if (data.error) {
+        // updateSnackBarMessage(data.error.message);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDescription.tsx]: Remove comment

File: client/src/pages/Board/CardDialogComponents/CardDescription.tsx [hunk 21]
```diff
@@ -0,0 +1,111 @@
...
+      } else if (data.success) {
+        updateBoards();
+      } else {
+        // should not get here from backend but this catch is for an unknown issue
+        console.error({ data });
+
+        // updateSnackBarMessage('An unexpected error occurred. Please try again');
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDescription.tsx]: Also remove this one

File: client/src/pages/Board/CardDialogComponents/CardDescription.tsx [hunk 22]
```diff
@@ -0,0 +1,111 @@
...
+    setIsSubmitting(false);
+  };
+
+  const handleDeleteDescription = async (): Promise<void> => {
+    deleteCardItem('description', newDescription, ids).then((data) => {
+      setIsSubmitting(true);
+      if (data.error) {
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        // updateCardContext(data.success);
```
Comment (ezeilo-su) [inline client/src/pages/Board/CardDialogComponents/CardDescription.tsx]: This comment is not needed. Please remove.

File: server/controllers/board.js [hunk 23]
```diff
@@ -11,6 +11,17 @@ const {
...
+exports.userBoard = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const userBoards = await User.findById(userId);
+  if (userBoards) {
+    return res.status(200).json({ boards: userBoards.boards });
+  }
+  res.status(500);
+  throw new Error("Something went wrong");
+});
```
Comment (ezeilo-su) [inline server/controllers/board.js:23]: Did you combine two tickets here? This one is a backend code.

**Review by ezeilo-su — CHANGES_REQUESTED** [review]
Hey @uche-okoroafor, you've done good work here. I've left you a couple of comments to address, though :smile:

**Comment by ezeilo-su** [comment]
Please resolve this merge conflict

**Review by moffatethan — APPROVED** [review]

**Pull request merged**